### DISCUSSION TITLE
Updated docs for generated Go client

### DISCUSSION THREAD

**Discussion by kenji-fukasawa** [choria-io/general#52] [Q&A]
Hello,

I just generated a new Go client using the `choria` utility.  However, it looks like the generated code doesn't match up with the [existing documentation](https://choria.io/docs/development/mcorpc/goclients/).  For example, it looks like the generated action functions now include all of the inputs as parameters.  That seems self-explanatory, but I'm a little hung up on the new syntax for `Must`.

```
func Must(fw inter.Framework, opts ...InitializationOption) (client)
```

Can you provide some information on what I should be providing for `fw`?

**Comment by ripienaar** [comment]
It's an instance of `choria.Framework` that provides auth, connections etc

**Reply by kenji-fukasawa** [reply]
Understood.  I think I've sorted out how to initialize the client.  What is the current method for filtering?  Is there any documentation or examples I can reference?

**Reply by ripienaar** [reply]
Discovery is still like in the docs here: https://choria.io/docs/development/mcorpc/goclients/

If you are making a CLI tool there are helpers to add the standard flags and do discovery based on them see https://github.com/choria-io/go-choria/blob/main/scout/cmd/trigger.go

Generated clients have a number of NodeSource implementations https://github.com/choria-io/go-choria/blob/main/client/scoutclient/discover.go
